chia/_tests/core/ssl/test_ssl.py

```diff
@@ -65,19 +104,62 @@ async def test_farmer(self, farmer_one_harvester, self_hostname):
+        _, farmer_service, bt = farmer_one_harvester
+        farmer_api = farmer_service._api
+
+        farmer_server = farmer_api.farmer.server
+        chia_ca_crt_path, chia_ca_key_path = chia_ssl_ca_paths(bt.root_path, bt.config)
+
+        #
+        # Use client certificate signed by public CA (instead of private one) and use public CA for SSL context
+        # This reliably raises ClientConnectorCertificateError
+        #
         pub_crt = farmer_server.root_path / "non_valid.crt"
         pub_key = farmer_server.root_path / "non_valid.key"
         generate_ca_signed_cert(chia_ca_crt_path.read_bytes(), chia_ca_key_path.read_bytes(), pub_crt, pub_key)
         ssl_context = ssl_context_for_client(chia_ca_crt_path, chia_ca_key_path, pub_crt, pub_key)
         with pytest.raises(aiohttp.ClientConnectorCertificateError):
             await establish_connection(farmer_server, self_hostname, ssl_context)
-        ssl_context = ssl_context_for_client(ca_private_crt_path, ca_private_key_path, pub_crt, pub_key)
-        with pytest.raises(aiohttp.ServerDisconnectedError):
-            await establish_connection(farmer_server, self_hostname, ssl_context)
+
+    @pytest.mark.anyio
+    async def test_farmer_mismatch_context(self, farmer_one_harvester_not_started, self_hostname, caplog):
+        _, farmer_service, bt = farmer_one_harvester_not_started
+        farmer_api = farmer_service._api
+
+        farmer_server = farmer_api.farmer.server
+        ca_private_crt_path, ca_private_key_path = private_ssl_ca_paths(bt.root_path, bt.config)
+        chia_ca_crt_path, chia_ca_key_path = chia_ssl_ca_paths(bt.root_path, bt.config)
+        #
+        # Use client certificate generated from public CA but use private CA for SSL context
+        # This doesn't reliable raise a specific exception on all platforms so resorting to searching
+        # the log for certificate failed messages. However, this is complicated in that the log
+        # for this is only generated with asyncio in debug mode which due to further complications
+        # needs to be set near the beginning when the farmer service starts
+        #
+        with set_asyncio_debug():
+            async with farmer_service.manage():
+                pub_crt = farmer_server.root_path / "non_valid.crt"
+                pub_key = farmer_server.root_path / "non_valid.key"
+                generate_ca_signed_cert(chia_ca_crt_path.read_bytes(), chia_ca_key_path.read_bytes(), pub_crt, pub_key)
+
+                ssl_context = ssl_context_for_client(ca_private_crt_path, ca_private_key_path, pub_crt, pub_key)
+                try:
+                    caplog.clear()
+                    with ignore_ssl_cert_error(), caplog.at_level(logging.DEBUG, logger="asyncio"):
+                        await establish_connection(farmer_server, self_hostname, ssl_context)
+                except Exception:
+                    pass  # ignore any exceptions and just check the expected log output below
```

Comment:
I think we want some failure here, even if we don't maintain a list of the possible exceptions.

```suggestion
                caplog.clear()
                with ignore_ssl_cert_error(), caplog.at_level(logging.DEBUG, logger="asyncio"):
                    with pytest.raises():
                        await establish_connection(farmer_server, self_hostname, ssl_context)
```

Reply:
changed per suggestion